Assets/MRTK/SDK/Experimental/Joystick/JoystickController.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Joystick
+{
+    /// <summary>
+    /// Example script to demonstrate joystick control in sample scene
+    /// </summary>
+    public class JoystickController : MonoBehaviour
+    {
+        [SerializeField]
+        [Tooltip("The large or small game object that receives manipulation by the joystick.")]
+        private GameObject objectToManipulate = null;
+
+        public GameObject ObjectToManipulate
+        {
+            get => objectToManipulate;
+            set => objectToManipulate = value;
+        }
+
+        [SerializeField]
+        [Tooltip("A TextMeshPro object that displays joystick values.")]
+        private TextMeshPro debugText = null;
+
+        [SerializeField]
+        [Tooltip("The joystick mesh that gets rotated when this control is interacted with.")]
+        private GameObject joystickVisual = null;
+
+        [SerializeField]
+        [Tooltip("The mesh + collider object that gets dragged and controls the joystick visual rotation.")]
+        private GameObject grabberVisual = null;
+
+        [SerializeField]
+        [Tooltip("Toggles on / off the GrabberVisual's mesh renderer because it can be dragged away from the joystick visual, it kind of breaks the illusion of pushing / pulling a lever.")]
+        private bool showGrabberVisual = true;
+
+        [SerializeField]
+        [Tooltip("The speed at which the JoystickVisual and GrabberVisual move / rotate back to a neutral position.")]
+        [Range(1, 20)]
+        private int reboundSpeed = 5;
+
+        [SerializeField]
+        [Tooltip("How sensitive the joystick reacts to dragging left / right. Customize this value to get the right feel for your scenario.")]
+        [Range(50, 300)]
+        private int sensitivityLeftRight = 100;
+
+        [SerializeField]
+        [Tooltip("How sensitive the joystick reacts to pushing / pulling. Customize this value to get the right feel for your scenario.")]
+        [Range(50, 300)]
+        private int sensitivityForwardBack = 150;
+
+        public enum JoystickMode { Move, Rotate, Scale }
+        public JoystickMode mode = JoystickMode.Move;
```

Comment:
Does this need to be public? If so, it also needs to follow the patterns and casing from https://github.com/microsoft/MixedRealityToolkit-Unity/pull/7978#discussion_r434252167

Reply:
With the public methods below for changing joystick mode (like `JoystickMode_Move`), I'm not sure this needs to also be public.